interfaces/builtin/custom_device.go

```diff
@@ -451,8 +462,33 @@ func (iface *customDeviceInterface) UDevConnectedPlug(spec *udev.Specification,
 	}
 
 	// Now write all the rules
-	for _, rule := range deviceRules {
-		spec.TagDevice(rule)
+	for deviceName, rule := range deviceRules {
+		if rule == placeholderRule {
+			baseName := filepath.Base(deviceName)
+
```

Comment:
maybe this would be a bit more readable if we separated the cases deviceName == baseName and deviceName != baseName more clearly?

Reply:
Yes, I like this too. The `deviceName == baseName` case is really simple, just emit rule, so I'll have it do that and then continue, leaving the remaining codepath for the `!=`, which can then be simpler.